### PR TITLE
Add event property types for macOS swift

### PR DESCRIPTION
**Pull request by charmian01**
## Description
Add event property types for macOS swift
Property types: String, Double, Long, Boolean, DateTime

### CODE REVIEW

File: SasquatchMac/SasquatchMac/ViewControllers/AnalyticsViewController.swift [hunk 1]
```diff
@@ -102,109 +130,97 @@ class AnalyticsViewController : NSViewController, NSTableViewDataSource, NSTable
...
 
   @IBAction func deleteProperty(_ : AnyObject) {
-    if properties.isEmpty {
-      return
-    }
-    guard let `table` = table else {
-      return
+    if let selectedProperty = arrayController.selectedObjects.first as? EventProperty {
+      arrayController.removeObject(selectedProperty)
+      selectedProperty.removeObserver(self, forKeyPath: #keyPath(EventProperty.type), context: nil)
```
Comment (ElektrojungeMS) [inline SasquatchMac/SasquatchMac/ViewControllers/AnalyticsViewController.swift]: Much cleaner! Nicely done!

File: SasquatchMac/SasquatchMac/ViewControllers/AnalyticsViewController.swift [hunk 2]
```diff
@@ -102,109 +130,97 @@ class AnalyticsViewController : NSViewController, NSTableViewDataSource, NSTable
...
+      let key = property.key
+      guard let type = EventPropertyType(rawValue: property.type) else {
+        continue
       }
-      if row < 0 {
-        return
+      switch type {
+      case .String:
+        properties.setEventProperty(property.string, forKey: key);
+        propertyDictionary[property.key] = (property.string )
```
Comment (MatkovIvan) [inline SasquatchMac/SasquatchMac/ViewControllers/AnalyticsViewController.swift]:
```suggestion
        propertyDictionary[property.key] = property.string
```

File: SasquatchMac/SasquatchMac/ViewControllers/AnalyticsViewController.swift [hunk 3]
```diff
@@ -3,9 +3,24 @@ import Cocoa
...
+    var type: String = EventPropertyType.String.rawValue
+    var string: String = ""
+    var double: NSNumber = 0
+    var long: NSNumber = 0
+    var boolean: Bool = false
+    var dateTime: Date = Date.init()
+  }
+
+  enum EventPropertyType : String {
+    case String = "String"
```
Comment (achocron) [inline SasquatchMac/SasquatchMac/ViewControllers/AnalyticsViewController.swift]: Can you please make the enums camel cased? (not the string values though)
so `string`, `double`, `long`, `boolean`, and `dateTime`?

Comment (charmian01) [reply SasquatchMac/SasquatchMac/ViewControllers/AnalyticsViewController.swift]: Fixed this in new commit.

File: SasquatchMac/SasquatchMac/ViewControllers/AnalyticsViewController.swift [hunk 4]
```diff
@@ -90,121 +118,109 @@ class AnalyticsViewController : NSViewController, NSTableViewDataSource, NSTable
 
   @IBAction func priorityChanged(_ sender: NSComboBox) {
     switch (self.priorityValue.stringValue) {
```
Comment (MatkovIvan) [inline SasquatchMac/SasquatchMac/ViewControllers/AnalyticsViewController.swift]:
```suggestion
     switch (Priority(rawValue: self.priorityValue.stringValue)!) {
```

Comment (charmian01) [reply SasquatchMac/SasquatchMac/ViewControllers/AnalyticsViewController.swift]: Resolved in new commit.

File: SasquatchMac/SasquatchMac/ViewControllers/AnalyticsViewController.swift [hunk 5]
```diff
@@ -89,122 +117,110 @@ class AnalyticsViewController : NSViewController, NSTableViewDataSource, NSTable
...
 
   @IBAction func priorityChanged(_ sender: NSComboBox) {
-    switch (self.priorityValue.stringValue) {
-    case Priority.Normal.rawValue:
-      self.priority = Priority.Normal
-    case Priority.Critical.rawValue:
-      self.priority = Priority.Critical
-    case Priority.Invalid.rawValue:
-      self.priority = Priority.Invalid
+    switch (Priority(rawValue: self.priorityValue.stringValue)!) {
```
[the remark this answers is not in the document]
Comment (charmian01) [reply SasquatchMac/SasquatchMac/ViewControllers/AnalyticsViewController.swift]: Yeah, that's brief. I have updated in new commit.

**Pull request merged**